lib/utils/ReporterStats.js

```diff
@@ -159,8 +159,12 @@ class ReporterStats extends RunnableStats {
 
     getTestStats (runner) {
         const suiteStats = this.getSuiteStats(runner, runner.parent)
-        if (!suiteStats.tests[runner.title]) throw Error(`Unrecognised test [${runner.title}] for suite [${runner.parent}]`)
-        return suiteStats.tests[runner.title]
+
+        // Errors encountered inside hooks (e.g. beforeEach) can be identified by looking
+        // at the currentTest param
+        const title = runner.currentTest || runner.title
```

Comment:
can you also specify that runner.currentTest is only specified in the mocha adapter?